### PR TITLE
Most past gen items should still be illegal

### PR DESCRIPTION
**Pull request by HoeenCoder**
Not sure if this is the best way to do this so PRed.

### CODE REVIEW

File: data/rulesets.js [hunk 1]
```diff
@@ -717,6 +717,14 @@ let BattleFormats = {
 	natdex: {
 		effectType: 'Rule',
 		name: 'NatDex',
+		onValidateSet(set) {
+			// Item's other than Mega Stones, Z-Crystals, and the Orbs should still be illegal.
```
Comment (KrisXV) [inline data/rulesets.js]: Items

Comment (Zarel) [reply data/rulesets.js]: Should they though

**Comment by Zarel** [comment]
Should they though

**Comment by KrisXV** [comment]
https://www.smogon.com/forums/threads/national-dex-mechanics-thread.3656786/ @Zarel read point 8
( Aside from Mega Stones, Z-Crystals, and Primal Orbs, no old items will be made legal (type gems, etc.))

**Comment by Zarel** [comment]
Did you misread? It explicitly says type gems are also legal, and more information is coming soon.

**Comment by Zarel** [comment]
Oh, huh, it was edited 11 minutes ago and I had the old version.

**Comment by Zarel** [comment]
Items that can be associated with a Pokémon probably should be. The teambuilder needs it, too.

**Comment by Zarel** [comment]
I talked it over with the NatDex tier leaders; the new list is:

> All Z-Crystals, Mega Stones, Arceus Plates (except Pixie Plate which is already in the game), and Genesect Drives, Blue Orb, Red Orb, Deep Sea Scale, Deep Sea Tooth, Lucky Punch, Soul Dew, Thick Club, Lustrous Orb, Adamant Orb, Griseous Orb

File: data/rulesets.js [hunk 2]
```diff
@@ -717,6 +717,14 @@ let BattleFormats = {
...
+			if (!set.item) return;
+			let item = this.dex.getItem(set.item);
+			if (item.isNonstandard === 'Past' && !item.megaStone && !item.zMove && !['redorb', 'blueorb'].includes(item.id)) {
```
Comment (Zarel) [inline data/rulesets.js]:
```suggestion
			if (item.isNonstandard === 'Past' && !item.zMove && !item.itemUser) {
```

(`item.itemUser` is a new field I'm introducing to the item database)

File: data/rulesets.js (hunk 1 above)
Comment (Zarel) [reply data/rulesets.js]:
```suggestion
			// Items other than Z-Crystals and Pokémon-specific items should be illegal
```

**Pull request merged**